braindecode/models/eegsym.py

```diff
@@ -0,0 +1,756 @@
+            ),
+        )
+
+        # Temporal reduction
+        self.temporal_reduction = nn.Sequential(
+            TemporalBlock(
+                in_channels=int(self.filters_per_branch * len(self.scales_samples) / 4),
+                filters=int(self.filters_per_branch * len(self.scales_samples) / 4),
+                kernel_size=4,
+                activation=self.activation,
+                drop_prob=self.drop_prob,
+            ),
+            nn.AvgPool3d(kernel_size=(1, 2, 1)),
+        )
+
+        # Channel merging
+        self.channel_merging = ChannelMergingBlock(
+            in_channels=int(self.filters_per_branch * len(self.scales_samples) / 4),
+            filters=int(self.filters_per_branch * len(self.scales_samples) / 4),
+            groups=int(self.filters_per_branch * len(self.scales_samples) / 8),
+            activation=self.activation,
+            drop_prob=self.drop_prob,
+        )
+
+        # Temporal merging
+        self.temporal_merging = TemporalMergingBlock(
+            in_channels=int(self.filters_per_branch * len(self.scales_samples) / 4),
+            filters=int(self.filters_per_branch * len(self.scales_samples) / 2),
+            groups=int(self.filters_per_branch * len(self.scales_samples) / 4),
+            activation=self.activation,
+            drop_prob=self.drop_prob,
+        )
+
+        # Output layers
+        self.output_blocks = nn.Sequential(
+            OutputBlock(
+                in_channels=int(self.filters_per_branch * len(self.scales_samples) / 2),
+                activation=self.activation,
+                drop_prob=self.drop_prob,
+            ),
+            nn.Flatten(),
+        )
+
+        # Final fully connected layer
+        self.final_layer = nn.Linear(
+            in_features=int(
+                int(self.filters_per_branch * len(self.scales_samples) / 2) * 2
+            ),
+            out_features=self.n_outputs,
+        )
+
+    def _create_inception_block(
+        self,
+        in_channels: int,
+        scales_samples: List[int],
+        filters_per_branch: int,
+        ncha: int,
+        average_pool: int,
+        init: bool = False,
+    ):
+        return InceptionBlock(
+            in_channels=in_channels,
+            scales_samples=scales_samples,
+            filters_per_branch=filters_per_branch,
+            ncha=ncha,
+            activation=self.activation,
+            drop_prob=self.drop_prob,
+            average_pool=average_pool,
+            spatial_resnet_repetitions=self.spatial_resnet_repetitions,
+            init=init,
+        )
+
+    def _create_residual_block(
+        self, in_channels: int, filters: int, kernel_size: int, average_pool: int
+    ):
+        return ResidualBlock(
+            in_channels=in_channels,
+            filters=filters,
+            kernel_size=kernel_size,
+            activation=self.activation,
+            drop_prob=self.drop_prob,
+            average_pool=average_pool,
+        )
```

Comment:
I created theses functions, but now know I don't if we need this extra function just to create the layers. What do you think @PierreGtch?

Reply:
Maybe not necessary indeed, it added a step when navigating the file.